src/Engine/AbstractSocketIO.php

```
@@ -102,6 +102,22 @@ public function emit($event, array $args)
         throw new UnsupportedActionException($this, 'emit');
     }
 
+    /**
+     * Network safe \fread wrapper
+     *
+     * @param integer $bytes
+     * @return bool|string
+     */
+    protected function readBytes($bytes)
+    {
+        $data = '';
+        while ($bytes > 0 && false !== ($chunk = \fread($this->stream, $bytes))) {
+            $bytes -= \strlen($chunk);
+            $data .= $chunk;
+        }
+        return $chunk === false ? false : $data;
```

instead of returning false, how about throwing an exception instead ?

---

yep that makes sense. Are you ok with that exception?